public/colorizer.ts

```diff
@@ -0,0 +1,92 @@
+import { Color, DataTexture, FloatType, RGBAFormat, RedFormat, RedIntegerFormat, LinearFilter, UnsignedByteType } from "three";
+
+function getSquarestTextureDimensions(size: number): [number, number] {
+  const width = Math.ceil(Math.sqrt(size));
+  const height = Math.ceil(size / width);
+
+  return [width, height];
+}
+
+function loadColormap(colorStops: string[]): DataTexture {
+  const colorColorStops = colorStops.map((color) => new Color(color));
+  const dataArr = colorColorStops.flatMap((col) => [col.r, col.g, col.b, 1]);
+  const colormapTex = new DataTexture(new Float32Array(dataArr), colorColorStops.length, 1, RGBAFormat, FloatType);
+  // if (this.type === ColorRampType.HARD_STOP) {
+  //   this.texture.minFilter = NearestFilter;
+  //   this.texture.magFilter = NearestFilter;
+  // } else {
+  colormapTex.minFilter = LinearFilter;
+  colormapTex.magFilter = LinearFilter;
+  // }
+  colormapTex.internalFormat = "RGBA32F";
+  colormapTex.needsUpdate = true;
+
+  return colormapTex;
+}
```

Comment:
We should be okay without the hard stop color ramps I think, because categorical colors _should_ be sampling directly at the color stops on the ramp.

Reply:
this code change is only in the example app here, and I only left it out due to laziness.  The creation of the DataTexture for colormaps is really TFE's responsibility and should work just fine, I think.